src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseParticipationResource.java

```diff
@@ -99,19 +95,33 @@ public ProgrammingExerciseParticipationResource(ProgrammingExerciseParticipation
             @RequestParam(defaultValue = "false") boolean withSubmission) {
         var participation = participationRepository.findByIdElseThrow(participationId);
         if (!programmingExerciseParticipationService.canAccessParticipation((ProgrammingExerciseParticipation) participation)) {
-            return forbidden();
+            throw new AccessForbiddenException("participation", participationId);
         }
 
         Optional<Result> result = resultRepository.findLatestResultWithFeedbacksForParticipation(participation.getId(), withSubmission);
         if (result.isPresent() && !authCheckService.isAtLeastTeachingAssistantForExercise(participation.getExercise())) {
-            final boolean isBeforeDueDate = participation.getExercise().isBeforeDueDate();
-            result.get().filterSensitiveInformation();
-            result.get().filterSensitiveFeedbacks(isBeforeDueDate);
+            filterResult(participation, result.get());
         }
 
         return result.map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.ok(null));
     }
 
+    /**
+     * Removes sensitive information that students should not see (yet) from the given result.
+     * @param participation the result belongs to.
+     * @param result the sensitive information of which should be removed.
+     */
+    private void filterResult(final Participation participation, final Result result) {
```

Comment:
I would rename this method, e.g. filterSensitiveInformationInResult

Reply:
:+1: Done in 207454ec9c77b6eefb1fd83a04884f7acd42f4de.